cloudify_agent/installer/__init__.py

```diff
@@ -387,9 +391,30 @@ def upload_certificate(self):
         cert_content = self.rest_cert_content
 
         if not self.runner.exists(cert_file):
+            cert_dir = os.path.dirname(cert_file)
+            if not self.runner.exists(cert_dir):
+                self._create_cert_dir(cert_dir)
+
             with tempfile.NamedTemporaryFile(delete=False) as f:
                 f.write(cert_content)
             try:
                 self.runner.put_file(f.name, cert_file)
             finally:
                 os.unlink(f.name)
+
+    def _create_cert_dir(self, cert_dir):
+        """Create the directory containing the manager certificate.
+
+        For cross-platform compatibility, use python to create the directory.
+        """
+
+        try:
+            self.runner.python(
```

Comment:
this is very ugly but it was the only way i could think of that didnt make me lose my sanity, re:

  - use mkdir on linux and whatever else (?) on windows
  - needs to create nested dirs (like `mkdir -p`)
  - needs to handle already-existing directories
  - needs to handle the `~`